### PR TITLE
feat(service): Add SendGrid service

### PR DESCRIPTION
**Pull request by chanioxaris**
Added support for SendGrid email service.

### CODE REVIEW

File: service/sendgrid/sendgrid.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+// New returns a new instance of a SendGrid notification service.
+// You will need a SendGrid API key.
+// See https://sendgrid.com/docs/for-developers/sending-email/api-getting-started/
+func New(apiKey, senderAddress, senderName string) *SendGrid {
+	client := sendgrid.NewSendClient(apiKey)
+
+	return &SendGrid{
+		client:            client,
```
Comment (NotVinay) [inline service/sendgrid/sendgrid.go:27]: nit: could drop client variable and assign directly here.

File: service/sendgrid/sendgrid.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+func (s SendGrid) Send(subject, message string) error {
+	from := mail.NewEmail(s.senderName, s.senderAddress)
+	c := mail.NewContent("text/html", message)
+
+	// Create a new personalization instance to be able to add multiple receiver addresses.
+	p := mail.NewPersonalization()
+	p.Subject = subject
+
+	for _, receiverAddress := range s.receiverAddresses {
+		receiverEmail := mail.NewEmail(receiverAddress, receiverAddress)
```
Comment (NotVinay) [inline service/sendgrid/sendgrid.go:51]: same here you could get rid of receiverEmail variable reduces unnecessary variable assignments, and keeps the code compact.

**Review by NotVinay — APPROVED** [review]

File: service/sendgrid/sendgrid.go (hunk 1 above)
Comment (chanioxaris) [reply service/sendgrid/sendgrid.go]: Thanks for pointing this out, applied your suggestion on a new commit

File: service/sendgrid/sendgrid.go (hunk 2 above)
Comment (chanioxaris) [reply service/sendgrid/sendgrid.go]: Thanks for pointing this out, applied your suggestion on a new commit

**Review by nikoksr — COMMENTED** [review]
Sweet, appreciate your contribution! Just some code style changes and we're ready to merge.

File: service/sendgrid/sendgrid.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+// a given message to all those addresses.
+func (s *SendGrid) AddReceivers(addresses ...string) {
+	s.receiverAddresses = append(s.receiverAddresses, addresses...)
+}
+
+// Send takes a message subject and a message body and sends them to all previously set chats. Message body supports
+// html as markup language.
+func (s SendGrid) Send(subject, message string) error {
+	from := mail.NewEmail(s.senderName, s.senderAddress)
+	c := mail.NewContent("text/html", message)
```
[the remark this answers is not in the document]
Comment (chanioxaris) [reply service/sendgrid/sendgrid.go]: Applied the requested changes

**Review by nikoksr — APPROVED** [review]

**Pull request merged**